presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -87,6 +88,9 @@
     private final LoadingCache<String, Set<String>> rolesCache;
     private final LoadingCache<PrestoPrincipal, Set<RoleGrant>> roleGrantsCache;
 
+    private boolean isPartitionVersioningEnabled;
+    private Map<HivePartitionName, Integer> partitionVersionMap;
```

Comment:
Given partition cache is a loading cache, this map might be outdated. Could we store this information in partition cache and use `getIfPresent`? They might achieve similar goal.